### ISSUE TITLE
Consumption directory /Users/me/PAPERLESS_CONSUMPTION does not exist

### ISSUE THREAD

**Issue by githubber** [danielquinn/paperless#67]
Hi!
I've got this setup using Vagrant. The web server runs, the settings I've set in settings.py for user and password work. BUT, when I run /opt/paperless/src/manage.py document_consumer I get a warning that it can't find my PAPERLESS_CONSUMPTION_DIR. It is in my Home folder, I've set it to be read/writeable by all, etc. What am I doing wrong?? Thank you, this seems like awesome software.

**Comment by danielquinn** [comment]
I don't typically use the Vagrant setup (that was contributed) but I'll spin up an instance and see if I can reproduce your issue.  I just made some changes to the names of some constants, but that *shouldn't* have affected Vagrant.  Perhaps I was wrong.  Stand by.

**Comment by danielquinn** [comment]
So, I poked around the vagrant instance I spun up and I couldn't reproduce the problem.  This made me think that it was something to do with whatever you did in `settings.py`, so I reworked the Vagrantfile and provisioning script so that people shouldn't have to poke around in there.  I've got it setup now that you just need to edit `/etc/paperless.conf` and put whatever values you need in there.

That's the good news.  The bad news is that this means that the vagrant box you've currently got won't work and you'll have to rebuild it after doing a `git pull` on this repo to acquire the updates.  I've updated the [documentation](http://paperless.readthedocs.org/en/latest/setup.html#setup-installation-vagrant) though, so at least there's that.

TL;DR:

```
git pull
vagrant destroy -f
vagrant up
(follow the instructions for editing your config at /etc/paperless.conf
```

Remember that once you edit `/etc/paperless.conf` you'll have to exit the vagrant box and ssh back in to update your environment.  After that, you should be good to go.

**Comment by githubber** [comment]
Hm, I am still getting an error like, "CommandError: Consumption directory /Users/me/PAPERLESS_CONSUMPTION does not exist". But when I first opened the paperless.conf I noticed that the suggested directory seemed to be inside the vagrant vm. I am really not even sure of the terminology. Does the directory have to be 'inside' the vm, or can it be anywhere? If anywhere, what syntax do I use? Thanks for the help. Did I mention I am doing this on a Mac, OS X 10.10.5?

**Comment by danielquinn** [comment]
Ah, now I see the problem.  Here's the thing: if Paperless is operating in a VM, then the documents it consumes must be visible from that VM in the form of a seemingly local directory.

That means that in order for you to get PDFs into a Vagrant-based Paperless system, you need to do one of the following:

* Share a directory between the VM (guest) and your main machine (host) systems and designate the path to that mount point in the VM's `paperless.conf` file.
* Copy the PDF files into the VM

There's an easy way to do this via the Vagrantfile, but I couldn't think of a path that's going to exist on everyone's machine, so I haven't setup this share by default.  Of course you can edit the Vagrantfile to do whatever you like for your own purposes.

Does any of this help?

**Comment by githubber** [comment]
Thank you. I hd tried sharing a local directory through VirtualBox, but setting it in the vagrantfile did the trick. Running /opt/paperless/src/manage.py document_consumer now starts without warnings. BUT, when I add a PDF to the local directory, nothing shows up in the web view. and there is no feedback in the Terminal following /opt/paperless/src/manage.py document_consumer. Is there a debug mode I can use?

**Comment by danielquinn** [comment]
A few things to know about the consumer:

* It checks the directory on a 10second loop, so while you may have placed your document in there, there might not be any action for as much as 10seconds.
* It operates as a daemon, constantly looking for documents in the directory you've designated.  There is default output though, so simply putting a file in there should result in some output, typically starting with something like `Consuming /home/ftp/Scans/my-file.pdf`.  This output is also logged to the database, so you should be able to see these messages by going into the admin and visiting `/documents/log/`.
* Actually processing a PDF can take anywhere from 30seconds to 10minutes, depending on the size and complexity of the document as well as the resources available to your server, so you can't expect documents to show up in the web view immediately.

If you aren't seeing any output, then there's a good chance that there's a mismatch between where you're putting the files and where Paperless is looking for them.  Try logging into the Vagrant instance and locating the file in the consumption directory.  If it's there, make sure that it (and the consumption directory itself) is readable by the `vagrant` user.

---

Also, I should point out that the quick fix I did this morning with the use of `paperless.conf` is likely to change later today.  It's the nature of bleeding edge projects I suppose.  Basically the file currently looks like this:

```
export PAPERLESS_CONSUMPTION_DIR="..."
export PAPERLESS_PASSPHRASE="..."
(and so on)
```

I'll be removing the `export ` bit soon, so it'll look like this:

```
PAPERLESS_CONSUMPTION_DIR="..."
PAPERLESS_PASSPHRASE="..."
(and so on)
```

I'm telling you this just so you aren't surprised if you do a `git pull` on this repo and suddenly things are exploding.  This change I did this morning was bit of a hurried hack 'cause I had to get to work :-/

**Comment by danielquinn** [comment]
Alright, that heads-up I told you about? It's here now.  If you do a `git pull` you'll also need to edit `/etc/paperless.conf` as explained above.  Of course let's get you sorted with the setup you've got first, so let me know if you need any more help.

**Comment by danielquinn** [comment]
Since there hasn't been any action on this for more than a day, I'm gonig to assume that you're all set now and close it.  If there's still something wrong though, feel free to re-open.
